### ISSUE TITLE
When using min&max in the datepicker, show year and multi-year view only if its needed.

### ISSUE THREAD

**Issue by artamigo** [angular/material2#9930]
#### Bug, feature request, or proposal:

Proposal for enhancement of datepicker

#### What is the expected behavior?

When using `minDate` and `maxDate`, optionally display the `year` or `multi-year` views if there is only one choice to be selected.

#### What is the current behavior?

The `Year` and `multi-year` views are displayed even though there is only one option to be selected.

https://stackblitz.com/edit/angular-8oy79a

<img width="312" alt="multy-year" src="https://user-images.githubusercontent.com/1064001/36164535-9a11c75a-10ba-11e8-9b41-90769f836b70.png">

<img width="308" alt="year" src="https://user-images.githubusercontent.com/1064001/36164527-93da60fe-10ba-11e8-8e86-c34ae8e6bee1.png">

#### What are the steps to reproduce?
Providing a StackBlitz reproduction is the *best* way to share your issue. <br/>
StackBlitz starter: https://goo.gl/wwnhMV<br/>

#### What is the use-case or motivation for changing an existing behavior?

With the existing behavior, the users have to click multiple times just to learn the there are no more options to be selected from, which is a bit confusing. 

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

5.2.0

#### Is there anything else we should know?

No.

**Comment by julianobrasil** [comment]
This is likely to be solved as soon as PR #9639 is finished. The developer will be able to customize the header.

**Comment by mmalerba** [comment]
Agreed that that this would be nice, the custom header can be used to work around it for now, thus marking as P4

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
